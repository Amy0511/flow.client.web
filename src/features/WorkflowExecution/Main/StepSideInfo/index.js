import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import getHumanizedDuration from "@boomerang/boomerang-utilities/lib/getHumanizedDuration";
import { ACTIVITY_STATUSES_TO_TEXT, ACTIVITY_STATUSES_TO_ICON } from "Constants/activityStatuses";
import "./styles.scss";

const StepSideInfo = ({ step }) => {
  return (
    <div className="c-step-side-info">
      <div className="s-step-side-info-title">Summary</div>
      <div className="c-step-side-info-details">
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Name</div>
          <div className="b-step-side-info-field__value">{step.taskName}</div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Type</div>
          <div className="b-step-side-info-field__value">{"Task"}</div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Status</div>
          <div className="b-step-side-info-field__value">
            {ACTIVITY_STATUSES_TO_TEXT[step.flowTaskStatus]}
            <img src={ACTIVITY_STATUSES_TO_ICON[step.flowTaskStatus]} alt="Status" />
          </div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Start Time</div>
          <div className="b-step-side-info-field__value">{moment(step.startTime).format("YYYY-MM-DD h:mm A")}</div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Duration</div>
          <div className="b-step-side-info-field__value">
            {step.duration ? getHumanizedDuration(step.duration) : "-----"}
          </div>
        </div>
      </div>
    </div>
  );
};

StepSideInfo.propTypes = {
  step: PropTypes.object.isRequired
};

export default StepSideInfo;
