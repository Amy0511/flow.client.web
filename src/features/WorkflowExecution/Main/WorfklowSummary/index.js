import React from "react";
import PropTypes from "prop-types";
import "./styles.scss";

const WorkflowSummary = ({ workflowData, version }) => {
  return (
    <div className="c-workflow-summary">
      <div className="s-workflow-summary-title">{workflowData.name || "Workflow"}</div>
      <div className="c-workflow-summary-details">
        <div className="b-workflow-summary-field">
          <div className="b-workflow-summary-field__key">Description</div>
          <div className="b-workflow-summary-field__value">{workflowData.shortDescription || ""}</div>
        </div>
        <div className="b-workflow-summary-field">
          <div className="b-workflow-summary-field__key">Version</div>
          <div className="b-workflow-summary-field__value">{version}</div>
        </div>
      </div>
    </div>
  );
};

WorkflowSummary.propTypes = {
  workflowData: PropTypes.object,
  version: PropTypes.number
};

export default WorkflowSummary;
