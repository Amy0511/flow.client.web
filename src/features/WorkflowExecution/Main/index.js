import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import LoadingAnimation from "@boomerang/boomerang-components/lib/LoadingAnimation";
import NavigateBack from "Components/NavigateBack";
import TimeProgressBar from "Components/TimeProgressBar";
import DiagramApplication from "Utilities/DiagramApplication";
import { DiagramWidget } from "@boomerang/boomerang-dag";
import StepSideInfo from "./StepSideInfo";
import WorkflowSummary from "./WorfklowSummary";
import "./styles.scss";

class Main extends Component {
  static propTypes = {
    dag: PropTypes.object.isRequired,
    taskId: PropTypes.string,
    workflowData: PropTypes.object.isRequired,
    workflowExecutionData: PropTypes.object.isRequired,
    updateActiveNode: PropTypes.func,
    version: PropTypes.number
  };

  constructor(props) {
    super(props);
    this.diagramApp = new DiagramApplication({ dag: props.dag, modelIsLocked: true });
  }

  render() {
    const { workflowExecutionData, taskId, updateActiveNode } = this.props;
    const hasStepExecuting =
      workflowExecutionData.steps && workflowExecutionData.steps.find(step => step.flowTaskStatus !== "notstarted");
    const selectedStep =
      workflowExecutionData.steps && workflowExecutionData.steps.length && taskId
        ? workflowExecutionData.steps.find(step => step.taskId === taskId)
        : undefined;

    return (
      <div className="c-workflow-execution">
        <nav style={{ marginBottom: "1rem", width: "15rem" }}>
          <NavigateBack
            to={this.props.location.state.from || "/activity"}
            text={`Back to ${this.props.location.state.fromText || "Activity"}`}
          />
        </nav>
        <TimeProgressBar updateActiveNode={updateActiveNode} tasks={workflowExecutionData} />
        <WorkflowSummary workflowData={this.props.workflowData} version={this.props.version} />
        {selectedStep && <StepSideInfo step={selectedStep} />}
        <div className="c-workflow-diagram-execution">
          {hasStepExecuting ? (
            <DiagramWidget
              className="c-diagram-canvas"
              diagramEngine={this.diagramApp.getDiagramEngine()}
              maxNumberPointsPerLink={0}
              deleteKeys={[]}
              allowLooseLinks={false}
              allowCanvasTranslation={true}
              allowCanvasZoo={false}
            />
          ) : (
            <LoadingAnimation theme="bmrg-white" message="Your workflow will be with you shortly" />
          )}
        </div>
      </div>
    );
  }
}

export default withRouter(Main);
