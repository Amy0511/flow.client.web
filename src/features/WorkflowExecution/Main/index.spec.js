import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Main from "./index";

jest.mock("Components/NavigateBack", () => "NavigateBack");

const mockfn = jest.fn();

const dag = {};
const taskId = "test";
const workflowData = {};
const workflowExecutionData = {};
const version = 1;

describe("Main --- Snapshot", () => {
  it("Capturing Snapshot of Main", () => {
    const renderedValue = renderer.create(
      <Main
        dag={dag}
        taskId={taskId}
        workflowData={workflowData}
        workflowExecutionData={workflowExecutionData}
        updateActiveNode={mockfn}
        version={version}
      />
    );
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("Main --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Main
        dag={dag}
        taskId={taskId}
        workflowData={workflowData}
        workflowExecutionData={workflowExecutionData}
        updateActiveNode={mockfn}
        version={version}
      />
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
