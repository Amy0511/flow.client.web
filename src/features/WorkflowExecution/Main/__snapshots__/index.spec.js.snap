// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main --- Snapshot Capturing Snapshot of Main 1`] = `
<div
  className="c-workflow-execution"
>
  <nav
    style={
      Object {
        "gridArea": "header",
        "marginBottom": "1rem",
        "width": "15rem",
      }
    }
  >
    <NavigateBack
      text="Back to Activity"
      to="/activity"
    />
  </nav>
  <div
    className="c-time-progress-bar"
  >
    <div
      className="b-time-progress-bar"
    >
      <div
        className="b-time-progress-bar__start"
      >
        Start
      </div>
      <div
        className="b-time-progress-bar__finish"
      >
        Finish
      </div>
      <div
        className="b-time-progress-bar__fillers"
      />
    </div>
  </div>
  <div
    className="c-workflow-diagram-execution"
  />
  <aside
    style={
      Object {
        "gridArea": "sidebar",
      }
    }
  >
    <div
      className="c-workflow-summary"
    >
      <div
        className="s-workflow-summary-title"
      >
        Workflow
      </div>
      <div
        className="c-workflow-summary-details"
      >
        <div
          className="b-workflow-summary-field"
        >
          <div
            className="b-workflow-summary-field__key"
          >
            Description
          </div>
          <div
            className="b-workflow-summary-field__value"
          >
            ---
          </div>
        </div>
        <div
          className="b-workflow-summary-field"
        >
          <div
            className="b-workflow-summary-field__key"
          >
            Version
          </div>
          <div
            className="b-workflow-summary-field__value"
          >
            1
          </div>
        </div>
        <div
          className="b-workflow-summary-field"
        >
          <div
            className="b-workflow-summary-field__key"
          >
            Status
          </div>
          <div
            className="b-workflow-summary-field__value"
          >
            Not Started
            <img
              alt="Status undefined"
              className="b-activity-card__status-icon"
              src="pending.svg"
            />
          </div>
        </div>
        <div
          className="b-workflow-summary-field"
        >
          <div
            className="b-workflow-summary-field__key"
          >
            Start Time
          </div>
          <div
            className="b-workflow-summary-field__value"
          >
            December 31st 2017, 7:00:00 pm
          </div>
        </div>
        <div
          className="b-workflow-summary-field"
        >
          <div
            className="b-workflow-summary-field__key"
          >
            Duration
          </div>
          <div
            className="b-workflow-summary-field__value"
          >
            ---
          </div>
        </div>
      </div>
    </div>
  </aside>
</div>
`;
